File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
#include <unordered_set>

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> myset;

        // Iterate through the vector
        for (int i=0; i<nums.size(); i++){
            // Check if the hash set has the number or not
            if (!myset.count(nums[i])){
                myset.insert(nums[i]);
            }
            else {
                return true;
            }
        }
        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
#Bucket sort solution

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        # We need a hash map to keep the occurence of each number
        count = {}
        # Initialize the bucket to the size of the array, which is just dict
        bucket = [[] for i in range(len(nums) + 1)]

        for i in nums:
            # count.get(i, 0) check for the value at key i
            # if no value found then return 0
            count[i] = 1 + count.get(i, 0)

        # Now we go through each number and its occurence to add to the list
        for i, c in count.items():
            # count is our index in the list
            bucket[c].append(i)

        top = []
        # Iterate through the list from the end
        # We store from smallest to highest occurence within the list
        for i in range(len(bucket)-1, 0, -1):
            for n in bucket[i]:
                top.append(n)
                if len(top) == k:
                    return top
